src/groups/mqb/mqbcmd/mqbcmd_parseutil.cpp

```diff
@@ -267,6 +291,41 @@ int parseCommand(Command* command, bsl::string* error, WordGenerator next)
     return -1;
 }
 
+// ENCODING ...
+int parseEncodingFormat(EncodingFormat::Value* format,
+                        bsl::string*           error,
+                        WordGenerator&         next)
+{
+    const bslstl::StringRef word = next();
+
+    const char errorCommon[] = "The ENCODING option must be followed by one "
+                               "of the [TEXT, JSON_COMPACT, JSON_PRETTY] "
+                               "settings (default: TEXT)";
+
+    if (word.empty()) {
+        *error += errorCommon;
+        *error += ", but no setting was specified.";
+        return -1;  // RETURN
+    }
+
+    if (equalCaseless(word, "TEXT")) {
+        *format = EncodingFormat::Value::TEXT;
+        return 0;  // RETURN
+    }
+    else if (equalCaseless(word, "JSON_COMPACT")) {
+        *format = EncodingFormat::Value::JSON_COMPACT;
+        return 0;  // RETURN
+    }
+    else if (equalCaseless(word, "JSON_PRETTY")) {
+        *format = EncodingFormat::Value::JSON_PRETTY;
```

Comment:
Possible to use `EncodingFormat::fromString`, but it is case sensitive. Probably we don't want to mix case sensitive `fromString` here with case insensitive plaintext admin commands.